aztec/src/main/kotlin/org/wordpress/aztec/watchers/DeleteMediaElementWatcher.kt

```diff
@@ -0,0 +1,46 @@
+package org.wordpress.aztec.watchers
+
+import android.text.Editable
+import android.text.TextWatcher
+import org.wordpress.aztec.AztecText
+import org.wordpress.aztec.Constants
+import org.wordpress.aztec.spans.AztecMediaSpan
+import java.lang.ref.WeakReference
+
+class DeleteMediaElementWatcher(aztecText: AztecText) : TextWatcher {
+
+    private val aztecTextRef: WeakReference<AztecText?> = WeakReference(aztecText)
+
+    private var deletedMedia: Boolean = false
+    private var mediaSpan: AztecMediaSpan? = null
+
+    override fun beforeTextChanged(text: CharSequence, start: Int, count: Int, after: Int) {
+        deletedMedia = count > 0 && text[start + count - 1] == Constants.IMG_CHAR
```

Comment:
nice catch - addressed in 8aec182. I'm now also considering there might be mutiple media items within the selected text the user is trying to delete, thus triggering each of their listeners as well.